### DISCUSSION TITLE
How to display a scrollable list in dropdown

### DISCUSSION THREAD

**Discussion by habics** [lapce/floem#382] [General]
How to make the list scrollable and set it to certain height?

![Untitled](https://github.com/lapce/floem/assets/161233868/0201e007-8674-4462-9782-85d774d018a1)

**Comment by jrmoulton** [comment]
Currently the dropdown list isn't wrapped in a scroll. The list view will need to be wrapped in a scroll and be targetable with a class. I'm throwing together a Pr to make this possible

**Reply by jrmoulton** [reply]
https://github.com/lapce/floem/pull/383

This will let you target the list by using the `widgets::ListClass` and the scroll by the `DropDownScrollClass`. Using those classes you can style both the list and the scroll as you normally would

**Comment by habics** [comment]
Thank you @jrmoulton 

I'll close this question as resolved once a minimal example with scrollable list is available here, to serve as a reference for others who might have the same question in the future.

**Comment by jrmoulton** [comment]
class styling is nestable. So an example styling might look like

```rust
let root_view = empty().style(|s| {
    s.class(widgets::dropdown::DropDownClass, |s| {
      s.items_center()
        .padding(5)
        .class(widgets::dropdown::DropDownScrollClass, |s| {
          s.min_height(0.)
            .flex_grow(1.)
            .flex_basis(0.)
            .height(100)
            .class(widgets::ListClass, |s| {
              s.class(widgets::ListItemClass, |s| {
                s.padding(5).margin_horiz(5).items_center()
              })
            })
        })
  })
});
```

I haven't actually tested that styling but that's the idea. Getting the flex properties for lists in scrolls is always a little tricky.
